addons/mail/static/src/js/chat_manager.js

```diff
@@ -761,6 +764,9 @@ var chat_manager = {
     get_needaction_counter: function () {
         return needaction_counter;
     },
+    get_starred_counter: function () {
+        return _.where(messages, {is_starred: true}).length || starred_counter;
```

Comment:
this is wrong.  If someone has 1 message starred in messages, then it will ignore the real number.  This method should just be "return starred_counter;"